tests/phpunit/tests/media.php

```diff
@@ -5390,6 +5390,36 @@ public function test_quality_with_image_conversion_file_sizes() {
 		}
 	}
 
+	/**
+	 * @ticket 62305
+	 */
+	public function test_image_meta_changes() {
+		$temp_dir = get_temp_dir();
+		$file     = $temp_dir . '/33772.jpg';
+		copy( DIR_TESTDATA . '/images/33772.jpg', $file );
+
+		$editor = wp_get_image_editor( $file );
+
+		// Verify that the selected editor supports WebP output.
+		if ( ! $editor->supports_mime_type( 'image/webp' ) ) {
+			$this->markTestSkipped( 'WebP is not supported by the selected image editor.' );
+		}
+
+		$attachment_id = self::factory()->attachment->create_object(
+			array(
+				'post_mime_type' => 'image/jpeg',
+				'file'           => $file,
+			)
+		);
+
+		// Generate all sizes as WebP.
+		add_filter( 'image_editor_output_format', array( $this, 'image_editor_output_webp' ) );
+		$webp_sizes = wp_generate_attachment_metadata( $attachment_id, $file );
+		remove_filter( 'image_editor_output_format', array( $this, 'image_editor_output_webp' ) );
+
+		$this->assertSame( $file, $webp_sizes['file'], 'The original file name should same.' );
```

Comment:
I think this should be `original_image`, as `file` is intended to be the full size version of the converted image.

Reply:
I agree, this comparison doesn't look right.

That said, I'm unclear what this test is covering. Per https://core.trac.wordpress.org/ticket/62305#comment:34, the Core change itself is not a bug. It does change the attachment metadata, but that's because of what it fixes (consistently convert the image if it requires conversion, regardless of whether it needs to also be scaled down or not).

Reply:
> I think this should be original_image, as file is intended to be the full size version of the converted image.
> ..
> It does change the attachment metadata, but that's because of what it fixes

Right. This is the actual bugfix from https://core.trac.wordpress.org/ticket/62305. Before https://core.trac.wordpress.org/changeset/59317 the uploaded image (i.e. the `file` from the image meta array) was not converted when conversion was required. After the fix it is converted.

This bug has existed for a while but became more visible/severe when converting .heic images as the originally uploaded image cannot be used in most browsers but WordPress was using it anyway.

Thinking this test can perhaps be changed to confirm that the original image is converted (when smaller than the big image threshold).

Reply:
Got it. Thanks for clarification 🙌

Reply:
My question is why the full size image generated with `33772-1920x1080.webp` not `33772.webp`?

When i upload https://github.com/WordPress/wordpress-develop/blob/trunk/tests/phpunit/data/images/33772.jpg image it generates below images:
- 33772-150x150.jpg
- 33772-300x169.jpg
- 33772-768x432.jpg
- 33772-1024x576.jpg
- 1536x864.jpg
- 33772.jpg

But when i convert JPEG to WEBP it generates below images:
- 33772-150x150.webp
- 33772-300x169.webp
- 33772-768x432.webp
- 33772-1024x576.webp
- 33772-1536x864.webp
- 33772-1920x1080.webp
- 33772.jpg

Reply:
Unit tests added in https://github.com/WordPress/wordpress-develop/pull/7733/commits/7c0e5e770e415376fa780c20f6007584b442f050

Reply:
@mukeshpanchal27 Thank you for clarifying. Indeed it's strange that the converted image receives the size suffix as if it was an image sub-size. Indeed that image, in your example, should be called `33772.webp` instead of `33772-1920x1080.webp`. This may point to a bug in Core.

In terms of unit tests, it would be great if you could expand the test to cover this problem - e.g. don't check just for the suffix, but check that the generated file name is the same as the uploaded file name, except `.jpg` replaced with `.webp`. This way you cover the finding you just shared. That unit test should then fail, and it will help us to validate a potential fix.

Reply:
@felixarntz Update in https://github.com/WordPress/wordpress-develop/pull/7733/commits/6106ef387130402bf448bac3bb45a9917cab8a93

Reply:
@felixarntz @mukeshpanchal27 The suffix is added as the suffix is undefined when passed to `WP_Image_Editor::generate_filename()`. It's not possible to pass `''` as the suffix due to a loose falsey check.

We could either remove it or use `-converted`, this allows for removal.

```diff
diff --git a/src/wp-admin/includes/image.php b/src/wp-admin/includes/image.php
index 2e79578e23..62af48cb08 100644
--- a/src/wp-admin/includes/image.php
+++ b/src/wp-admin/includes/image.php
@@ -341,6 +341,18 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 				 */
 				if ( $scale_down ) {
 					$saved = $editor->save( $editor->generate_filename( 'scaled' ) );
+				} elseif ( $convert ) {
+					/*
+					 * Generate a new file name for the converted image.
+					 *
+					 * As the image file name will be unique due to the changed file extension,
+					 * it does not need a suffix to be unique. However, the generate_filename method
+					 * does not allow for an empty suffix, so the "-converted" suffix is required to
+					 * be added and subsequently removed.
+					 */
+					$converted_file_name = $editor->generate_filename( 'converted' );
+					$converted_file_name = preg_replace( '/-converted\.[a-z0-9]+$/i', '.', $converted_file_name );
+					$saved = $editor->save( $converted_file_name );
 				} else {
 					$saved = $editor->save();
 				}
```

Reply:
@peterwilsoncc Something like `-converted` would be okay for me, though I would prefer to consider changing that internal logic - as in: If the file extension is different from the input image, it should be possible to keep using the same name (except the extension of course).

Reply:
I agree changing the logic would be optimal but I think other falsey values can end up being used, eg `::get_suffix()` can return `false`. So `::generate_filename( ::get_suffix() )` requires the loose check (I briefly considered using `false` to indicate no suffix) 

Using `-converted` is the smaller fix but WP would be stuck with it forever more, so I'm tempted to put in the snowflake code for now to allow for any niceties later. Either way I am happier than using the dimensions.

Reply:
Yes, for a moment I was considering to change `::generate_filename( ::get_suffix() )` to accept an empty string for no suffix. Didn't because of RC, but will open a ticket for 6.8.

Reply:
Hm, another option here may be to include the old file extension as a suffix? For example: converting `image.heic` to JPEG would produce `image-heic.jpg`?

Reply:
> Hm, another option here may be to include the old file extension as a suffix? For example: converting `image.heic` to JPEG would produce `image-heic.jpg`?

This is what we did in Modern Image Formats.